client/asset/eth/eth.go

```diff
@@ -1682,16 +1687,32 @@ func (w *ETHWallet) Swap(swaps *asset.Swaps) ([]asset.Receipt, asset.Coin, uint6
 		swapVal += contract.Value
 	}
 
-	oneSwap, _, _, err := w.swapGas(1, swaps.Version)
+	// Set the gas limit as high as reserves will allow.
+	n := len(swaps.Contracts)
+	oneSwap, nSwap, _, err := w.swapGas(n, swaps.Version, false) // allow live est to fail, using our own gas values
```

Comment:
If the live estimate fails here, and our hardcoded estimates are higher than the actual live estimates, then wouldn't `swapVal+fees > reservedVal` always be true?

Reply:
I thought they would be equal.  `reservedVal` should turn out to be `swapVal + g.Swap * n`.  Basing that on my understanding on how the reserves (via the `fundingCoin`) are initially set.

Reply:
It's generally a sign of major problems coming if the live estimates are higher than the hard-coded estimates.  In other words, it is (and has been) normal to have the hard coded estimates be what comes out of `swapGas`.
Consider the code on `master`, the checks below haven't changed, only added a fallback to the more realistic batched gas limits.